idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/inspection/InspectionUtilitySyntax.kt

```diff
@@ -16,15 +30,91 @@ interface InspectionUtilitySyntax {
     return listOf(expect) + actuals
   }
 
+  /**
+   * convenience function where [f] reduces FunctionReturnTypes of subsequent calls in the initializer to their returnType
+   */
+  fun <F : CallableDescriptor> F.returns(prop: KtProperty, types: KotlinBuiltIns.() -> List<KotlinType>): Boolean =
```

Comment:
This probably belongs in the KtUtils in the compiler-plugin module and not here unless these are IDE specific in their types and not accessible from there